### ISSUE TITLE
Getting "error":"not_authenticated" when i try to logout, Next 13 App Dir

### ISSUE THREAD

**Issue by Jared-Dahlke** [auth0/nextjs-auth0#1370]
### Checklist

- [X] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I copied the App Dir example from web-app_nextjs_javascript_hello_world to get authentication working.

I'm using 
```
 "next": "^13.4.16",
 "@auth0/nextjs-auth0": "^3.0.0",
```
I'm able to login and visit protected pages as expected. Refresh the browser , etc, no problems.

but when I call logout ( `/api/auth/logout`) i get an error that says on the window:
```
{"error":"not_authenticated","description":"The user does not have an active session or is not authenticated"}
```
when i look at the console logs i see some errors there too:

```
Access to fetch at 'https://kosm-dev-verses.us.auth0.com/v2/logout?client_id=redactedW6vbeVLQSoxLi3zX&returnTo=http%3A%2F%2Flocalhost%3A3000' (redirected from 'http://localhost:3000/api/auth/logout?_rsc=3vrfr') from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

GET https://kosm-dev-verses.us.auth0.com/v2/logout?client_id=redactedbeVLQSoxLi3Mn9KPI07EpzX&returnTo=http%3A%2F%2Flocalhost%3A3000 net::ERR_FAILED

Failed to fetch RSC payload. Falling back to browser navigation. TypeError: Failed to fetch
    at fetchServerResponse (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/router-reducer/fetch-server-response.js:51:27)
    at navigateReducer (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/router-reducer/reducers/navigate-reducer.js:161:123)
    at clientReducer (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/router-reducer/router-reducer.js:25:61)
    at eval (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/use-reducer-with-devtools.js:63:21)
    at updateReducerImpl (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:11221:22)
    at updateReducer (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:11103:10)
    at Object.useReducer (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:12521:16)
    at useReducer (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react/cjs/react.development.js:1761:21)
    at useReducerWithReduxDevtoolsImpl (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/use-reducer-with-devtools.js:102:53)
    at Router (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/app-router.js:164:169)
    at renderWithHooks (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:10697:18)
    at updateFunctionComponent (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:15180:20)
    at beginWork$1 (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:17324:16)
    at beginWork (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:25675:14)
    at performUnitOfWork (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24526:12)
    at workLoopConcurrent (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24512:5)
    at renderRootConcurrent (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24468:9)
    at performConcurrentWorkOnRoot (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:23339:38)
    at workLoop (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:261:34)
    at flushWork (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:230:14)
    at MessagePort.performWorkUntilDeadline (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:534:21)
window.console.error @ app-index.js:31
console.error @ hydration-error-info.js:45
fetchServerResponse @ fetch-server-response.js:94
await in fetchServerResponse (async)
navigateReducer @ navigate-reducer.js:164
clientReducer @ router-reducer.js:25
eval @ use-reducer-with-devtools.js:63
updateReducerImpl @ react-dom.development.js:11221
updateReducer @ react-dom.development.js:11103
useReducer @ react-dom.development.js:12521
useReducer @ react.development.js:1760
useReducerWithReduxDevtoolsImpl @ use-reducer-with-devtools.js:102
Router @ app-router.js:174
renderWithHooks @ react-dom.development.js:10697
updateFunctionComponent @ react-dom.development.js:15180
beginWork$1 @ react-dom.development.js:17324
beginWork @ react-dom.development.js:25675
performUnitOfWork @ react-dom.development.js:24526
workLoopConcurrent @ react-dom.development.js:24512
renderRootConcurrent @ react-dom.development.js:24468
performConcurrentWorkOnRoot @ react-dom.development.js:23339
workLoop @ scheduler.development.js:261
flushWork @ scheduler.development.js:230
performWorkUntilDeadline @ scheduler.development.js:534
     GET http://localhost:3000/api/auth/logout 401 (Unauthorized)

```

This is my app/api/auth/[auth0]:
```
import { paths } from "@/paths";
import { handleAuth, handleLogin } from "@auth0/nextjs-auth0";

export const GET = handleAuth({
  login: handleLogin({
    authorizationParams: {
      prompt: "login",
      
    },
    returnTo: paths.WORKSPACES,
  }),
  
  signup: handleLogin({
    authorizationParams: {
      prompt: "login",
      screen_hint: "signup",
    },
    returnTo: paths.WORKSPACES,
  }),
});

```

This is the last thing keeping me from using App Dir! Please help.
Any tips appreciated.
I've tried using various methods to try to force a logout

```
const GET = async (req: Request) => {
  const res = new NextResponse();

  cookies().delete("appSession");

  //  return redirect("/");

  return NextResponse.json({ test: "123" });
};

export { GET };
```

and a few other things.

Just need to figure out a way to let the user logout. 

PS> after posting this i also realized i am also getting an error when i login (go to /api/auth/login):
```
Access to fetch at 'https://kosm-dev-verses.us.auth0.com/authorize?client_id=redactedvbeVLQSoxLi3Mn9KPI07EpzX&scope=openid%20profile%20email&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fauth%2Fcallback&prompt=login&nonce=CsX1II021eX5E2PtTuFE0V3kMFl8V0H46IiLYImyHtQ&state=eyJyZXR1cm5UbyI6Ii9hcHAvd29ya3NwYWNlcyJ9&code_challenge=vVCdOBStaT2aXIqwZcWthlpPvwZKoeIw-Ov2tZQ_1dk&code_challenge_method=S256' (redirected from 'http://localhost:3000/api/auth/login?_rsc=acgkz') from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
fetch-server-response.js:61     GET https://kosm-dev-verses.us.auth0.com/authorize?client_id=redactedeVLQSoxLi3Mn9KPI07EpzX&scope=openid%20profile%20email&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fauth%2Fcallback&prompt=login&nonce=CsX1II021eX5E2PtTuFE0V3kMFl8V0H46IiLYImyHtQ&state=eyJyZXR1cm5UbyI6Ii9hcHAvd29ya3NwYWNlcyJ9&code_challenge=vVCdOBStaT2aXIqwZcWthlpPvwZKoeIw-Ov2tZQ_1dk&code_challenge_method=S256 net::ERR_FAILED
fetchServerResponse @ fetch-server-response.js:61
navigateReducer @ navigate-reducer.js:164
clientReducer @ router-reducer.js:25
eval @ use-reducer-with-devtools.js:63
updateReducerImpl @ react-dom.development.js:11221
updateReducer @ react-dom.development.js:11103
useReducer @ react-dom.development.js:12521
useReducer @ react.development.js:1760
useReducerWithReduxDevtoolsImpl @ use-reducer-with-devtools.js:102
Router @ app-router.js:174
renderWithHooks @ react-dom.development.js:10697
updateFunctionComponent @ react-dom.development.js:15180
beginWork$1 @ react-dom.development.js:17324
beginWork @ react-dom.development.js:25675
performUnitOfWork @ react-dom.development.js:24526
workLoopConcurrent @ react-dom.development.js:24512
renderRootConcurrent @ react-dom.development.js:24468
performConcurrentWorkOnRoot @ react-dom.development.js:23339
workLoop @ scheduler.development.js:261
flushWork @ scheduler.development.js:230
performWorkUntilDeadline @ scheduler.development.js:534
app-index.js:31 Failed to fetch RSC payload. Falling back to browser navigation. TypeError: Failed to fetch
    at fetchServerResponse (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/client/components/router-reducer/fetch-server-response.js:51:27)
    at navigateReducer (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/client/components/router-reducer/reducers/navigate-reducer.js:161:123)
    at clientReducer (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/client/components/router-reducer/router-reducer.js:25:61)
    at eval (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/client/components/use-reducer-with-devtools.js:63:21)
    at updateReducerImpl (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:11221:22)
    at updateReducer (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:11103:10)
    at Object.useReducer (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:12521:16)
    at useReducer (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react/cjs/react.development.js:1761:21)
    at useReducerWithReduxDevtoolsImpl (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/client/components/use-reducer-with-devtools.js:102:53)
    at Router (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/client/components/app-router.js:164:169)
    at renderWithHooks (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:10697:18)
    at updateFunctionComponent (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:15180:20)
    at beginWork$1 (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:17324:16)
    at beginWork (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:25675:14)
    at performUnitOfWork (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24526:12)
    at workLoopConcurrent (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24512:5)
    at renderRootConcurrent (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:24468:9)
    at performConcurrentWorkOnRoot (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:23339:38)
    at workLoop (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:261:34)
    at flushWork (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:230:14)
    at MessagePort.performWorkUntilDeadline (webpack-internal:///(/u/app-pages-browser)/./node_modules/next/dist/compiled/scheduler/cjs/scheduler.development.js:534:21)
```

### Reproduction

1. try to logout using /api/auth/logout
2. observe the errors that i pasted in the description

### nextjs-auth0 version

"@auth0/nextjs-auth0": "^3.0.0",

### Next.js version

"next": "^13.4.16"

### Node.js version

20

**Comment by Jared-Dahlke** [comment]
I realized i need to use bare <a> tags instead of link which fixed the cors errors.

Now my problem is when i log in, it redirects me back to  index/login page for some reason. i am having to login twice

**Comment by Jared-Dahlke** [comment]
this specific problem is solved. now my problem is after login it takes me back to index/login page. i will open a separate issue for this.
